storage/internal/benchmarks/client_pool.go

```diff
@@ -156,20 +174,41 @@ func getClient(ctx context.Context, api benchmarkAPI) *storage.Client {
 // mutex on starting a client so that we can set an env variable for GRPC clients
 var clientMu sync.Mutex
 
-func initializeHTTPClient(ctx context.Context, writeBufferSize, readBufferSize int, json bool) (*storage.Client, error) {
+// Client config
+type clientConfig struct {
+	writeBufferSize, readBufferSize int
+	useJSON                         bool // only applicable to HTTP Clients
+	setGCSFuseOpts                  bool // only applicable to HTTP Clients
+	connectionPoolSize              int  // only applicable to GRPC Clients
+}
+
+func initializeHTTPClient(ctx context.Context, config clientConfig) (*storage.Client, error) {
 	opts := []option.ClientOption{}
 
-	if writeBufferSize != useDefault || readBufferSize != useDefault {
+	if config.writeBufferSize != useDefault || config.readBufferSize != useDefault || config.setGCSFuseOpts {
 		// We need to modify the underlying HTTP client
-
 		base := http.DefaultTransport.(*http.Transport).Clone()
-		base.MaxIdleConnsPerHost = 100 // this is set in Storage as well
-		base.WriteBufferSize = writeBufferSize
-		base.ReadBufferSize = readBufferSize
 
-		http2Trans, err := http2.ConfigureTransports(base)
-		if err == nil {
-			http2Trans.ReadIdleTimeout = time.Second * 31
+		if config.setGCSFuseOpts {
+			base = &http.Transport{
```

Comment:
We don't need to create another object, since we are cloning the transport object in line:190. We can just override the attributes. Like,
base.MaxConnsPerHost = 100
base.MaxIdleConnsPerHost = 100
base.TLSNextProto = make(map[string]func(string, *tls.Conn) http.RoundTripper)

Reply:
I'd rather keep the exact initialization code as gcsFuse has. `http.DefaultTransport` configures several transport variables to default values, and I'm not sure manually initializing an `http.Transport` would have those configured the same.

Reply:
Okay.